src/ts/components/dashboard/TransactionList.tsx

```diff
@@ -11,25 +11,30 @@ import { displayAddress } from '../../services/address-transformer'
 interface ITransactionListProps extends StateProps, DispatchProps, RouteComponentProps {}
 
 interface ITransactionListState {
-  currentAddress: string
+  currentAddress: string,
+  currentNetwork: string
 }
 
 class TransactionList extends React.Component<ITransactionListProps, ITransactionListState> {
 
   state = {
-    currentAddress: ''
+    currentAddress: '',
+    currentNetwork: ''
   }
 
   static getDerivedStateFromProps (nextProps, prevState) {
     if (nextProps.account && nextProps.account.address !== prevState.currentAddress) {
       return { currentAddress: nextProps.account.address }
+    } else if (nextProps.network !== prevState.currentNetwork) {
+      return { currentAddress: nextProps.network }
```

Comment:
should this be currentNetwork?

Reply:
good catch

Reply:
I just fixed it since Robin appears offline, so that I can merge this branch